File: QytetetJava/src/modeloqytetet/Casilla.java
/**
 * @author Elena Cantero Molina
 */
package modeloqytetet;

public abstract class Casilla {
    protected int numeroCasilla;
    protected int coste;  
    //private int numHoteles; 
    //private int numCasas;   
    protected TipoCasilla tipo;
    //private TituloPropiedad titulo;
    
    public Casilla(int numeroCasilla, int coste, TipoCasilla tipo){
        this.numeroCasilla = numeroCasilla;
        this.coste = coste;
        this.tipo = tipo;
    }    
    
    /*public Casilla(int numeroCasilla, int coste, TipoCasilla tipo, TituloPropiedad titulo){
        this.numeroCasilla = numeroCasilla;
        this.coste = coste;
        this.numCasas = 0;
        this.numHoteles = 0;
        this.tipo = tipo;
        this.titulo = titulo;
    }*/
        
    public int GetNumeroCasilla(){ return this.numeroCasilla; }

    public int GetCoste(){ return this.coste; }

    public TipoCasilla GetTipo(){ return this.tipo; }  

    boolean soyEdificable(){ return this.GetTipo() == TipoCasilla.CALLE; }    

    private void asignarTituloPropiedad(){
    }

    @Override
    public String toString() {
        return "Casilla -> " + numeroCasilla + "\ncoste -> " + coste + "\ntipo -> " + tipo + '\n';
    }
}

File: QytetetRuby/lib/casilla.rb
#Elena Cantero Molina
#encoding: utf-8
#trig.rb

module ModeloQytetet
  class Casilla
    attr_accessor :numero_casas, :numero_hoteles, :titulo, :numero_casilla
    attr_reader :tipo, :coste
  
    def initialize(numero_casilla, coste, tipo) 
      @numero_casilla = numero_casilla
      @coste = coste
      @tipo = tipo
    end
    
    def soy_edificable
      return @tipo.eql?(TipoCasilla::CALLE)
    end    

    def asignar_titulo_propiedad      
    end
    
    def to_s
        "Casilla #{@numero_casilla}\nTipo de Casilla -> #{@tipo}\nCoste -> #{@coste}"
    end
  end
end
